server/configHelpers.js

```diff
@@ -0,0 +1,185 @@
+const { resolve } = require('path');
+const Config = require('bcfg');
+
+const logger = require('./logger');
+const clientFactory = require('./clientFactory');
+const { loadConfig, getConfigFromOptions } = require('./loadConfigs');
+
+/*
+ * create client config
+ * Note: This will actually create the file in your bpanel prefix location
+ * @param {string} id - id for the client
+ * @param {Object} options object for a bcoin/hsd client
+ * @param {bool} force - whether or not to force config creation if client
+ * can't connect
+ * @returns {bcfg.Config}
+ */
+async function createClientConfig(id, options = {}, force = false) {
+  assert(typeof id === 'string', 'Must pass an id as first paramater');
+  // assert(options, 'Must pass config options to create a client');
+
+  let clientConfig = options;
+  if (!(options instanceof Config))
+    clientConfig = getConfigFromOptions({ id, ...options });
+
+  const appConfig = loadConfig('bpanel', options);
+  const clientsDir = appConfig.str('clients-dir', 'clients');
+
+  // get full path to client configs relative to the project
+  // prefix which defaults to `~/.bpanel`
+  const clientsPath = resolve(appConfig.prefix, clientsDir);
+
+  try {
+    await testConfigOptions(clientConfig);
+  } catch (e) {
+    if (force) {
+      logger.warn(e.message);
+      logger.warn('Creating config file anyway...');
+    } else throw e;
+  }
+
+  let configTxt = '';
+  for (let key in clientConfig.options) {
+    const configKey = key
+      .replace('-', '')
+      .replace('_', '')
+      .toLowerCase();
+    const text = `${configKey}: ${clientConfig.options[key]}\n`;
+    configTxt = configTxt.concat(text);
+  }
+  if (!fs.existsSync(clientsPath)) {
+    logger.warn(
+      `Could not find requested client directory at ${clientsPath}. Creating new one...`
+    );
+    fs.mkdirpSync(clientsPath);
+  }
+  fs.writeFileSync(`${clientsPath}/${clientConfig.str('id')}.conf`, configTxt);
+  return clientConfig;
+}
+
+/*
+ * Retrieve a config from clients directory
+ * @param {string} id - id of client to retrieve
+ * @returns {bcfg.Config} - bcfg object of config
+ */
+
+function getConfig(id) {
+  assert(typeof id === 'string', 'Client config must have an id');
+
+  const appConfig = loadConfig('bpanel');
+  const clientsDir = appConfig.str('clients-dir', 'clients');
+
+  const clientsPath = resolve(appConfig.prefix, clientsDir);
+  const config = loadConfig(id, { id, prefix: clientsPath });
+
+  const path = resolve(clientsPath, `${id}.conf`);
+  if (!fs.existsSync(path)) {
+    const error = new Error(`File ${id}.conf not found`);
+    error.code = 'ENOENT';
+    throw error;
+  }
+
+  config.open(`${id}.conf`);
+  return config;
+}
+
+/*
+ * Simple utility that deletes a config
+ * Does not throw errors if client is not found
+ * NOTE: This does not confirm. Deletion is final!
+ * @param {string} id
+ * @returns {bool} - returns true when operation completed successfully
+ */
+
+function deleteConfig(id) {
```

Comment:
Does this function return `true` no matter what?

Reply:
Yes, that's right. I was torn around this, but my thinking was that you could get some information about the server by giving more information in return. That said, we already have endpoints that tell you if something is there or not. What are your thoughts? should it return false if not deleted?

Reply:
The jsdoc says:
`@returns {bool} - returns true when operation completed successfully`

If it always returns true, it might as well not return anything because you get no useful introspection into what is happening in the function. It has a side effect and that is it

Reply:
We could also remove that `success` intermediate variable and just return `true` in the `try` and `false` in the `catch`

Reply:
Yeah, that's what I'm doing now 👍